tests/test_bonded.py

```diff
@@ -196,6 +196,59 @@ def test_harmonic_bond(self, n_particles=64, n_bonds=35, dim=3):
             np.testing.assert_array_equal(test_du_dp, test_du_dp_rev)
             np.testing.assert_array_equal(test_du_dl, test_du_dl_rev)
 
+    def test_flat_bottom_bond(self, n_particles=64, n_bonds=35, dim=3):
+        """Randomly connect pairs of particles, then validate the resulting FlatBottomBond force"""
+        np.random.seed(2022)
+
+        # TODO(deboggle) : reduce code duplication between HarmonicBond and FlatBottomBond
+        box = np.eye(3) * 100
+        x = self.get_random_coords(n_particles, dim)
+
+        atom_idxs = np.arange(n_particles)
+
+        # (k, r_min, r_max)
+        params = np.random.rand(n_bonds, 3).astype(np.float64)
+        params[:, 0] *= 1000  # make k large
+        params[:, -1] += params[:, -2]  # guarantee r_max >= r_min
```

Comment:
Is this the wrong way around? (looks like it guarantees `r_min >= r_max`)

Reply:
I think this is correct (initialize `r_min, r_max` non-negative, update `r_max += r_min`)

```python
import numpy as np
n_bonds = 100

params = np.random.rand(n_bonds, 3).astype(np.float64)
params[:, 0] *= 1000  # make k large
params[:, -1] += params[:, -2]
k, r_min, r_max = params.T
assert (r_max >= r_min).all()
```


but I should make this more direct...

Reply:
Ah, sorry -- I was reading `params[:, -1] += params[:, -2]` as ` params[:, 1] += params[:, 2]`

Reply:
Rephrased in https://github.com/proteneer/timemachine/pull/764/commits/e93b8b5df364716e4bfc42cce7e155ceaf6a129b